Update uInverter Release Files

mohamed-dek1 opened the pull request:
This PR updates the uInverter REV C release files to be standardized to be similar to the AMDC REV D files. The changes include:

- Updates file names to follow the dash-case [naming convention](https://github.com/Severson-Group/KnowledgeBase/blob/main/CONTRIBUTING.md#file-naming)
- Adds odb folder that contains intelligent ODB++ files for PC assembly
- Adds pcb-layers file that contains all the layers of the board with silkscreens if applicable
- Adds `pcb-3d` image
- Adds `pdf-3d` PDF
- Adds mechanical components to the Altium schematic library, schematic, and schematic PDF.
- Updates README.md to have the standard template material similar to the AMDS and AMDC
- Updates the BOM to be an Altium-generated one that contains the columns similar to the AMDS and AMDC

##
this pr closes #257

mohamed-dek1 commented:
Below are some screenshots of the reports used to validate the table entries in the `README.md`

##

![image](https://github.com/user-attachments/assets/58c3b3c4-d33c-4f7b-88fb-a326b1b73a4b)
![image](https://github.com/user-attachments/assets/a911d5d3-6e68-49cf-b81b-902ec492e4f5)


##

@elsevers, there is the Altium-generated BOM `altium.csv` and a modified version of it `uinv-rev-c-bom.csv`. Which one do you want me to keep with the folder? I modified it because it doesn't have any of the mechanical components used for the heatsink and MOSFET. The Altium-generated BOM also has an empty `Value` column, so I also populated that, among other smaller changes.

elsevers commented:
Thanks @mohamed-dek1. It would be best if we can configure Altium to generate a BOM with the value fields populated and mechanical components included. We do this on the AMDC and AMDS, right?

See the note [here](https://github.com/Severson-Group/KnowledgeBase/blob/main/PCB-Design-Guidelines.md#generating-bom) on how to include mechanical components on the BOM:

<img width="1062" alt="image" src="https://github.com/user-attachments/assets/de65e068-8eb6-4245-846d-a32c989af372" />

mohamed-dek1 commented:
I followed the instructions, but the `Value` column still appears to be empty. Because the Heatsink/MOSFET mechanical components were never added to the schematic, they don't show up at all. Only the screws and standoffs are showing up.

##

![image](https://github.com/user-attachments/assets/e5870a9a-6139-4c9d-a2ab-76d54708d9d5)

elsevers commented:
Okay, got it. Can you add the heat sink / screws to the schematic? 

Here are google search results on how to populate the value field:

<img width="1023" alt="image" src="https://github.com/user-attachments/assets/950416e6-ab0c-4d0a-b28f-255e195ad5ad" />

Our goal here should be to make this consistent with the AMDC / AMDS

mohamed-dek1 commented:
- Added the mechanical components to the uInverter schematic library so that anyone who clones this repo should have access to them.
- Populated the `Value` field for any components that needed it.
- Added the Altium-generated BOM that now has the `Value` column.

elsevers commented on altium/PCB_v6.txt:
This PR should not be trying to make changes to the Altium files for the AMDC. Can you fix this?

mohamed-dek1 replied:
reverted changes to altium folder

elsevers commented on altium/PCB_v6.html:
This PR should not be trying to make changes to the Altium files for the AMDC. Can you fix this?

mohamed-dek1 replied:
reverted changes to altium folder

elsevers commented on Accessories/ExpansionBoard_uInverter/REV20220921C/README.md:
```diff
@@ -1,29 +1,39 @@
+
+## Bill of Materials
+
+The [Bill of Materials (BOM)](uinv-rev-c-bom.csv) contains components for 1 board. If users of this repo discover that any components are unvailable, please maintain an alternate-components.md file in this folder that lists approved alternates.
```
```suggestion
The [Bill of Materials (BOM)](uinv-rev-c-bom.csv) contains components for 1 board. If users of this repo discover that any components are unavailable, please maintain an `alternate-components.md` file in this folder that lists approved alternates.
```

elsevers reviewed (changes requested):
This is looking good! Thanks @mohamed-dek1 

I have listed a couple of change requests in my review. In addition to this, can you please update the PR description to have a brief bulleted list of the high level changes this PR is making?

mohamed-dek1 commented:
Checked files on seperate computer

elsevers reviewed (changes requested):
Thanks @mohamed-dek1.

I am comparing the mechanical parts on your BOM to the [Google sheet we created for the hardware we will stock in ME 76](https://docs.google.com/spreadsheets/d/16eWoWGY_jfMSNuZQeauz20iACJenNDp30TCMVM3sQI8/edit?gid=1879693648#gid=1879693648) and am discovering discrepancies for the `uInverter`.

It seems that several of the parts we added to the [Google Sheet](https://docs.google.com/spreadsheets/d/16eWoWGY_jfMSNuZQeauz20iACJenNDp30TCMVM3sQI8/edit?gid=1879693648#gid=1879693648) for heat sink mounting do not match the BOM:

![image](https://github.com/user-attachments/assets/dee89c9d-7b0f-4031-ad79-2a3c066e7be8)

Please go through this and figure out what happened. Why are these different part numbers? Update your BOM and the Google Sheet to have the correct parts. Indicate in the comments how you know you have the right parts indicated in both places.

Here are the issues recording your efforts to figure out the heat sink mounting supplies that we should stock:
- https://github.com/Severson-Group/Lab/issues/245
- https://github.com/Severson-Group/Lab/issues/270
- #247
- #248

mohamed-dek1 commented:
After reviewing the [BOM](https://github.com/Severson-Group/AMDC-Hardware/blob/docs/uinverter-revc-standardization/Accessories/ExpansionBoard_uInverter/REV20220921C/uinv-rev-c-bom.csv) and the [Google Sheet](https://docs.google.com/spreadsheets/d/16eWoWGY_jfMSNuZQeauz20iACJenNDp30TCMVM3sQI8/edit?gid=1879693648#gid=1879693648), I realized that the sheet was referring to the part numbers in this image found in the following issue:

##
![image](https://github.com/user-attachments/assets/b67a0e63-0356-4787-b8b4-bc21ebea3bd8)

- https://github.com/Severson-Group/Lab/issues/270

The correct part numbers are linked below the image:

##
![image](https://github.com/user-attachments/assets/f6b227dc-a810-4396-9b8b-f1f5ccfca9da)

They are the same components, and the discrepancy is for convenience (ordering all components from one place). I've updated the sheet to reflect the correct components and location, similar to the BOM.

mohamed-dek1 commented:
checked embedded images on another computer

elsevers reviewed (approved):
Great job @mohamed-dek1. Let's `squash and merge`
